sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/ArrayOverlapOperatorConversion.java

```diff
@@ -149,6 +153,42 @@ public DimFilter toDruidFilter(
         );
       }
     }
+
+    // if the input is a direct array column, we can use sweet array filter
+    if (simpleExtractionExpr.isDirectColumnAccess() && simpleExtractionExpr.isArray()) {
+      // To convert this expression filter into an OR of ArrayContainsElement filters, we need to extract all array
+      // elements.
+      if (expr.isLiteral()) {
```

Comment:
what will happen if its an empty array?

Reply:
hmm, so i guess it will fail a precondition check when it makes the OR filter because fields will be empty...

However, it currently isn't really possible to make an empty array literal because `ARRAY[]` isn't accepted by calcite (or the native druid `array` expression), so I'm not really sure how important it is to handle this case either.

I can update this (and `ArrayContainsOperatorConversion`) to return `null` in this case since we don't really handle it, or true i guess would also probably be correct?